aya/src/maps/lpm_trie.rs

```diff
@@ -0,0 +1,456 @@
+//! A LPM Trie.
+use std::{convert::TryFrom, marker::PhantomData, mem, ops::Deref};
+
+use crate::{
+    generated::bpf_map_type::BPF_MAP_TYPE_LPM_TRIE,
+    maps::{IterableMap, Map, MapError, MapRef, MapRefMut},
+    sys::{bpf_map_delete_elem, bpf_map_lookup_elem, bpf_map_update_elem},
+    Pod,
+};
+
+/// A Longest Prefix Match Trie.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 4.20.
+///
+/// # Examples
+/// ```no_run
+/// # let bpf = aya::Bpf::load(&[])?;
+/// use aya::maps::lpm_trie::{LpmTrie, Key};
+/// use std::convert::TryFrom;
+/// use std::net::Ipv4Addr;
+///
+/// let mut trie = LpmTrie::try_from(bpf.map_mut("LPM_TRIE")?)?;
+/// let ipaddr = Ipv4Addr::new(8,8,8,8);
+/// // The following represents a key for the "8.8.8.8/16" subnet.
+/// // The 'prefix_len' field represents how many bytes should be matched against
+/// // and the data represents the actoual data to be matched.
+/// let key = Key::new(16, u32::from(ipaddr).to_be());
+/// trie.insert(&key, 1, 0)?;
+///
+/// // LPMTrie matches against the longest (most accurate) key.
+/// let lookup = Key::new(32, u32::from(ipaddr).to_be());
+/// let value = trie.get(&lookup, 0)?;
+/// assert!(matches!(value,1));
+///
+/// // If we were to insert a key with longer 'prefix_len'
+/// // our trie should match against it.
+/// let longer_key = Key::new(24, u32::from(ipaddr).to_be());
```

Comment:
I think you mean shorter? 24 < 32

Reply:
Read further - I used this longer key to emphasis the "longest prefix matching" property. 24 > 16

Reply:
Ok, but what is confusing is that you bind longer_key after the longest prefix (32). I think the example should show:

- insert prefix 16;
- get prefix 16 (straightforward)
- get prefix 24 (returns the key inserted with prefix 16)
- insert prefix 32
- get prefix 32 (returns key inserted with prefix 32)